controllers/managedocs_controller_test.go

```diff
@@ -2,639 +2,759 @@ package controllers
+				Expect(k8sClient.Delete(ctx, secret)).Should(Succeed())
 			})
 		})
+		When("there is a valid size in the add-on parameter secret", func() {
+			It("should create reconciled resources", func() {
+				// Create a valid add-on parameters secret
+				secret := addonParamsSecretTemplate.DeepCopy()
+				secret.Data["size"] = []byte("1")
+				Expect(k8sClient.Create(ctx, secret)).Should(Succeed())
 
-		When("there is no storage cluster resource in the cluster", func() {
-			It("should create a new storage cluster", func() {
-				ctx := context.Background()
+				By("Creating a storagecluster resource")
+				utils.WaitForResource(k8sClient, ctx, scTemplate.DeepCopy(), timeout, interval)
 
-				scList := &ocsv1.StorageClusterList{}
-				Expect(k8sClient.List(ctx, scList, client.InNamespace(TestNamespace))).Should(Succeed())
-				Expect(scList.Items).Should(HaveLen(0))
+				By("Creating a prometous resource")
+				utils.WaitForResource(k8sClient, ctx, promTemplate.DeepCopy(), timeout, interval)
 
-				// Create the secret
-				secret := &corev1.Secret{
-					ObjectMeta: metav1.ObjectMeta{
-						Name:      TestAddOnParamsSecretName,
-						Namespace: TestNamespace,
-					},
-					Data: map[string][]byte{
-						"size": []byte("1"),
-					},
-				}
-				Expect(k8sClient.Create(ctx, secret)).Should(Succeed())
-				Expect(k8sClient.Get(ctx, utils.GetResourceKey(secret), secret)).Should(Succeed())
+				By("Creating an alertmanager resource")
+				utils.WaitForResource(k8sClient, ctx, amTemplate.DeepCopy(), timeout, interval)
+			})
+		})
+		When("size is increased in the add-on parameters secret", func() {
+			It("should increase storagecluster's storage device set count", func() {
+				secret := addonParamsSecretTemplate.DeepCopy()
+				secret.Data["size"] = []byte("4")
+				Expect(k8sClient.Update(ctx, secret)).Should(Succeed())
 
-				managedOCS := &v1.ManagedOCS{
-					ObjectMeta: metav1.ObjectMeta{
-						Name:      ManagedOCSName,
-						Namespace: TestNamespace,
-					},
-				}
-				Expect(k8sClient.Get(ctx, utils.GetResourceKey(managedOCS), managedOCS)).Should(Succeed())
+				// wait for the storagecluster to update
+				Eventually(func() bool {
+					sc := scTemplate.DeepCopy()
+					err := k8sClient.Get(ctx, utils.GetResourceKey(sc), sc)
+					if err != nil {
+						return false
+					}
+
+					var ds *ocsv1.StorageDeviceSet = nil
+					for index := range sc.Spec.StorageDeviceSets {
+						item := &sc.Spec.StorageDeviceSets[index]
+						if item.Name == deviceSetName {
+							ds = item
+							break
+						}
+					}
+					return ds != nil && ds.Count == 4
+				}, timeout, interval).Should(BeTrue())
 
-				sc := &ocsv1.StorageCluster{
-					ObjectMeta: metav1.ObjectMeta{
-						Name:      storageClusterName,
-						Namespace: TestNamespace,
-					},
-				}
-				utils.WaitForResource(k8sClient, ctx, sc, timeout, interval)
 			})
 		})
+		When("size is decreased in the add-on parameters secret", func() {
```

Comment:
```suggestion
		When("size is decreased in the add-on parameter's secret", func() {
```

Reply:
It is not a single parameter, so it will be parameters` but If you think of add-on parameters as the name of the secret, it can just fit as it is

Reply:
That works; I haven't thought of it that way.